src/Planarity/is-plane-description.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    is_plane_description(g, g.fadjlist)
+
+Returns `true` if the given combinatorial description of graph `g` is plane.
+
+The algorithm requires the graph `g` to be simple and undirected.
+### Optional Arguments
+- `description = g.fadjlist`: if the description is specified, it uses the new description instead of using `g.fadjlist`.
+
+A drawing of a planar graph induces a cyclic clockwise order in the neighbors of each vertex.
+A list of these cyclic orders is called a combinatorial description of the drawing.
+A facial cycle of a combinatorial description D is a cycle (v_0,v_1,...,v_k,v_0) where v_i+1 succeeds v_i-1 in the list D(v_i).
+A combinatorial description of a plane graph is also called plane.
+Clearly, every planar graph has a combinatorial description which is plane.
+This algorithm is correct because of the following theorem:
+    A combinatorial description D of a drawing of a graph G is plane
+        if and only if
+    nv(G) - ne(G) + fD = 2, where fD is the number of facial cycles of description D.
+
+This algorithm is presented in the dissertation available at https://www.ime.usp.br/~coelho/sh/index.html.
+"""
+
+function is_plane_description end
+
+@traitfn function is_plane_description(g::AG::(!IsDirected),
+                    description::Vector{Vector{T}} 
+                    = g.fadjlist) where {T <: Integer, U, AG <: AbstractGraph{U}}
```

Comment:
This is probably not a good idea `fadjlist` is only defined for `SimpleGraphs` so your code will only work for them (and will produce errors for every other graph type). What exactly do you need to capture here, and can you do it via iterating over `outneighbors` instead?

Reply:
> This is probably not a good idea: `fadjlist` is only defined for `SimpleGraphs` so your code will only work for them (and will produce errors for every other graph type). What exactly do you need to capture here, and can you do it via iterating over `outneighbors` instead?

In this function we need a graph and its combinatorial descprition, as you can see, the descprition is a Vector{Vector{T}} and we also can consider a adjacency list of a graph as a descprition, so my thought behind that was to create a function that can receive specified descpritions by the user and, in case of no specified decision, it would get its adjacency list as a description.

Reply:
But not all graph types have an (easily accessible) adjacency list.

I don't know what a combinatorial description is, so I can't provide any further guidance except that using `g.fadjlist` is definitely a no-no. Even using the `fadj()` accessor will limit you tremendously.

Reply:
Does it even make sense, to provide a default initialization? Won't this in most cases not fit the definition of a "plane combinatorial description", meaning that that the algorithm will return false, even if the graph is planar?

Also, is it even possible that there are planar graphs where there is no permutation of the vertices of a that graph. such that the sorted adjacency list is a plane combinatorial description?

Reply:
> Does it even make sense, to provide a default initialization? Won't this in most cases not fit the definition of a "plane combinatorial description", meaning that that the algorithm will return false, even if the graph is planar?
> 
> Also, is it even possible that there are planar graphs where there is no permutation of the vertices of a that graph. such that the sorted adjacency list is a plane combinatorial description?

In my view, it wouldn't make so much sense to provide a deafult initialization, I do it because I think it would be nice in case that someone wants to know if the existing adjacency list is already a plane description even if the answer is usually false.
Yes, it is possible, the graph k_5 and k_3,3 for example, don't have a plane description. The function works in this way: if it returns "true" we know that the description is plane and if it returns "false" we know that this description is not plane, but we can't say that there is no combination that returns true.

Reply:
> But not all graph types have an (easily accessible) adjacency list.
> 
> I don't know what a combinatorial description is, so I can't provide any further guidance except that using `g.fadjlist` is definitely a no-no. Even using the `fadj()` accessor will limit you tremendously.
> 
> Can you make this parameter a function instead? That is, `outneighbors(g, v)` gives you what `g.fadjlist[v]` (which you use below) gives you, in an API-compatible way.

I feel like it would be better just to remove the default value, reading simon's comment I don't think anymore that this is a good idea.

Reply:
I`ve stopped using the default value even just working with SimpleGraphs (another change)